python/src/space/core/storage.py

```diff
@@ -77,12 +83,24 @@ def __init__(self, location: str, metadata_file: str,
         self._physical_schema)
 
     self._primary_keys = set(self._metadata.schema.primary_keys)
+    self._current_branch = current_branch
```

Comment:
Can we set self._current_branch to _MAIN_BRANCH, when current_branch is None? then we can make sure self._current_branch is always populated.